c/shared/source/uforead/uforead.c

```diff
@@ -1819,9 +1818,153 @@ static void reallocFDArray(ufoCtx h){
     h->top.FDArray.array = newFDArray;
 }
 
+static void parseXMLFile(ufoCtx h, char* filename, const char* filetype){
+    xmlDocPtr doc;
+    xmlNodePtr cur;
+    unsigned char* keyID;
+    char* keyName;
+
+    xmlKeepBlanksDefault(0);
+
+    h->cb.stm.xml_read(&h->cb.stm, h->stm.src, &h->src.buf, &doc);
+
+    if (doc == NULL) {
+        fatal(h, ufoErrParse, "Failed to parse '%s'.\n", "file");
+    }
+
+    cur = xmlDocGetRootElement(doc);
+    if (cur == NULL) {  // if document empty
+        xmlFreeDoc(doc);
+//        return(NULL);
+    }
+
+    if (!xmlStrEqual((cur)->name, (const xmlChar *) filetype)) {
+        fprintf(stderr, "document of the wrong type, root node != %s", filetype);
+        xmlFreeDoc(doc);
+//        return(NULL);
+    }
+
+    cur = (cur)->xmlChildrenNode;
+    while (cur && xmlIsBlankNode(cur)) {
+        cur = (cur) -> next;
+    }
+    if ( cur == 0 ) {
```

Comment:
Better to check pointers against NULL. Also, given that the return is commented out this seems unsafe -- we're about to use `cur->name` in a test.